packages/analytics-js-integrations/__tests__/integrations/XPixel/browser.test.js

```diff
@@ -0,0 +1,173 @@
+    window.twq = jest.fn();
+  });
+
+  test('send pageview', () => {
+    xPixel.page({
+      message: {
+        context: {},
+        properties: {
+          category: 'test cat',
+          path: '/test',
+          url: 'http://localhost',
+          referrer: '',
+          title: 'test page',
+          testDimension: 'abc',
+          value: 35.0,
+          currency: 'GBP',
+        },
+      },
+    });
+    expect(window.twq.mock.calls[0]).toEqual(['event', '456', { currency: 'GBP', value: 35 }]);
+    expect(window.twq.mock.calls[1]).toEqual(['event', '467', { currency: 'GBP', value: 35 }]);
+  });
+});
+
+describe('XPixel Track event', () => {
+  let xPixel;
+  beforeEach(() => {});
+  test('Testing Track Simple Event', () => {
+    xPixel = new XPixel(basicConfig, { loglevel: 'DEBUG' }, destinationInfo);
+    xPixel.init();
+    window.twq = jest.fn();
+    xPixel.track({
+      message: {
+        context: {},
+        event: 'Custom',
+        properties: {
+          customProp: 'testProp',
+          order_id: 'transactionId',
+          value: 35.0,
+          coupon: 'APPARELSALE',
+          currency: 'GBP',
+          products: [
+            {
+              customPropProd: 'testPropProd',
+              product_id: 'abc',
+              category: 'Merch',
+              name: 'Food/Drink',
+              brand: '',
+              variant: 'Extra topped',
+              price: 3.0,
+              quantity: 2,
+              currency: 'GBP',
+              position: 1,
+              value: 6.0,
+              typeOfProduct: 'Food',
+            },
+          ],
+        },
+      },
+    });
+    expect(window.twq.mock.calls[0]).toEqual([
+      'event',
+      '987',
+      {
+        currency: 'GBP',
+        value: 35,
+        contents: [
+          {
+            content_type: 'product',
+            content_id: 'abc',
+            content_name: 'Food/Drink',
+            price: 3,
+            num_items: 2,
+          },
+        ],
+      },
+    ]);
+  });
+
+  test('Testing Track product_added with no content_type in payload', () => {
+    xPixel = new XPixel(basicConfig, { loglevel: 'DEBUG' });
+    xPixel.init();
+    window.twq.track = jest.fn();
+    xPixel.track({
+      message: {
+        context: {},
+        event: 'product_added',
+        properties: {
+          customProp: 'testProp',
+          event_id: 'purchaseId',
+          order_id: 'transactionId',
+          value: 35.0,
+          currency: 'GBP',
+          products: [
+            {
+              customPropProd: 'testPropProd',
+              product_id: 'abc',
+              category: 'Merch',
+              name: 'Food',
+              price: 3.0,
+              quantity: 2,
+              currency: 'GBP',
+              position: 1,
+              value: 6.0,
+              typeOfProduct: 'Food',
+            },
+          ],
+        },
+      },
+    });
+    expect(window.twq.mock.calls[0]).toEqual(['config', '12567839']);
```

Comment:
This test case is wrong can't see any use case of it and additionally whatever ouput we are expecting is in itself wrong